polynote-frontend/polynote/ui/component/search.ts

```diff
@@ -77,6 +76,29 @@ export class SearchModal extends Modal implements IDisposable {
             })
 
         }).disposeWith(this);
+
+        // in case something goes wrong while searching, display an error message and update the "searching..." message
+        ErrorStateHandler.get.view("serverErrors").addObserver((errors) => {
+            if (errors.length === 0) { // search succeeded
+                this.searchStatus.classList.remove("limit-width");
+                return;
+            }
+            let message = "Something went wrong while searching:\n";
```

Comment:
Consider building the message as an array and adding this prefix at the end, only if the array isn't empty. I think that would be clearer and you wouldn't need the `hasParsingFailure` flag. 

You can use the [collect helper](https://github.com/polynote/polynote/blob/master/polynote-frontend/polynote/util/helpers.ts#L220) to do this in a functional way